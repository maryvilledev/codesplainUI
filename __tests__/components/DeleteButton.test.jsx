import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import EnhancedDeleteButton, {
  DeleteButton,
} from '../../src/components/buttons/DeleteButton';

const defaultProps = {
  onClick: jest.fn(),
  snippetKey: 'snippetKey',
  isEnabled: false,
};

describe('<DeleteButton />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <DeleteButton
        {...defaultProps}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: onClick', () => {
    it('is triggered when clicked', () => {
      const onClick = jest.fn();
      const wrapper = shallowWithContext(
        <DeleteButton
          onClick={onClick}
          isEnabled
        />,
    );
      const button = wrapper.find('IconButton');
      button.simulate('touchTap');
      expect(onClick).toHaveBeenCalled();
    });
  });

  it('renders nothing if user cannot delete this snippet', () => {
    const wrapper = shallowWithContext(
      <EnhancedDeleteButton
        {...defaultProps}
      />,
    );
    expect(wrapper.find('Nothing').length).toEqual(1);
  });

  it('renders nothing if snippet has not been saved', () => {
    const snippetKey = '';
    const wrapper = shallowWithContext(
      <EnhancedDeleteButton
        {...defaultProps}
        snippetKey={snippetKey}
      />,
    );
    expect(wrapper.find('Nothing').length).toEqual(1);
  });
});
