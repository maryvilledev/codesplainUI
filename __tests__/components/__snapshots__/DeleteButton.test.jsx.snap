exports[`<DeleteButton /> matches snapshot of when the app is in read-only mode 1`] = `
<span
  style={
    Object {
      "height": "48px",
    }
  }>
  <div
    data-for="trash-tip"
    data-tip={true}
    style={
      Object {
        "display": "inline-block",
      }
    }>
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      id="TrashButton"
      onTouchTap={[Function]}
      style={
        Object {
          "zIndex": 5,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}>
      <ActionDelete />
    </IconButton>
  </div>
  <ReactTooltip
    effect="solid"
    id="trash-tip"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div">
    <span>
      Click to delete snippet
    </span>
  </ReactTooltip>
</span>
`;

exports[`<DeleteButton /> matches snapshot of when the app is not in read-only mode 1`] = `
<span
  style={
    Object {
      "height": "48px",
    }
  }>
  <div
    data-for="trash-tip"
    data-tip={true}
    style={
      Object {
        "display": "inline-block",
      }
    }>
    <IconButton
      disableTouchRipple={false}
      disabled={true}
      iconStyle={Object {}}
      id="TrashButton"
      onTouchTap={[Function]}
      style={
        Object {
          "zIndex": 5,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}>
      <ActionDelete />
    </IconButton>
  </div>
  <ReactTooltip
    effect="solid"
    id="trash-tip"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div">
    <span>
      You can only delete snippets
      <br />
      that you own
    </span>
  </ReactTooltip>
</span>
`;
